Sources/SwiftDocC/Model/Rendering/RenderContentCompiler.swift

```diff
@@ -47,7 +47,41 @@ struct RenderContentCompiler: MarkupVisitor {
     
     mutating func visitCodeBlock(_ codeBlock: CodeBlock) -> [any RenderContent] {
         // Default to the bundle's code listing syntax if one is not explicitly declared in the code block.
-        return [RenderBlockContent.codeListing(.init(syntax: codeBlock.language ?? bundle.info.defaultCodeListingLanguage, code: codeBlock.code.splitByNewlines, metadata: nil))]
+
+        if FeatureFlags.current.isExperimentalCodeBlockAnnotationsEnabled {
+
+            func parseLanguageString(_ input: String?) -> (lang: String? , tokens: [RenderBlockContent.CodeListing.OptionName]) {
```

Comment:
non-blocking: In the long term it might be worth extracting this so that the warning and the rendering don't have behavioral differences. Since there's already follow up PRs that parse more things, it's probably better to extract this in those PRs rather than here.